sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryAvroUtilsTest.java

```diff
@@ -501,48 +813,4 @@ public void testSchemaCollisionsInAvroConversion() {
     String output = BigQueryAvroUtils.toGenericAvroSchema(schema, false).toString();
     assertThat(output.length(), greaterThan(0));
   }
-
-  /** Pojo class used as the record type in tests. */
-  @SuppressWarnings("unused") // Used by Avro reflection.
-  static class Bird {
```

Comment:
BigQueryAvroUtilsTest has been largely rewritten. Would you mind sharing the idea behind the newly formed tests.

Also, this POJO class is removed and I do not see a replacement in new tests. Are the related path (getSchema from POJO) still covered by tests?

Reply:
Working with this POJO was difficult due to the field names not reflecting the underlying type.

I reworked the test suite to test all conversion in isolation.
It's also more convenient to check the `useAvroLogicalType` parameter: In most cases, the converted schema is identical regardless of this value. we can reused the expected schema definition in both cases.

Some extra cases have been added like parameterized numeric